lib/core/HomeFeed.js

```diff
@@ -0,0 +1,91 @@
+const ResultsParser = require("../parser/contents");
+const Simplify = require("../parser/simplify");
+const Utils = require("../utils/Utils");
+
+class HomeFeed {
+    #page;
+    #videos;
+    #video_elements;
+    constructor(data) {
+        this.#page = ResultsParser.parseResponse(data);
+    }
+
+    /**
+     * Get the original page data
+     */
+    get page() {
+        return this.#page;
+    }
+
+    /**
+     * Get all the videos in the home feed
+     * @returns {Array<import('../parser/contents/Video')>}
+     */
+    getVideos() {
+        if (this.#video_elements) return this.#video_elements;
+        
+        this.#video_elements = Simplify.matching({
+            type: Simplify.matching(/^Video$/),
+        }).runOn(this.#page);
+
+        return this.#video_elements;
+    }
+
+    /**
+     * Get all the videos in the home feed
+     * @deprecated Use getVideos instead
+     */
+    get videos() {
+        if (this.#videos) return this.#videos;
+
+        const simplified = this.getVideos();
+
+        this.#videos = simplified.map(video => {
+            return {
+                id: video.id,
+                title: video.title.toString(),
+                description: video.description,
+                channel: {
+                    id: video.author.id,
+                    name: video.author.name,
+                    url: video.author.endpoint.metadata.url,
+                    thumbnail: video.author.best_thumbnail,
+                    is_verified: video.author.is_verified,
+                    is_verified_artist: video.author.is_verified_artist,
+                },
+                metadata: {
+                    view_count: video.views.toString(),
+                    short_view_count_text: {
+                        simple_text: video.short_view_count.toString(),
+                        // TODO: the accessiblity text is not yet parsed
+                        accessibility_label: "",
+                    },
+                    thumbnail: video.best_thumbnail,
+                    thumbnails: video.thumbnails,
+                    // XXX: It doesn't look like this is returned?
```

Comment:
It is, but not always. If you go through the results you'll eventually find at least one video with moving thumbnails.
Not sure why it behaves like that though.

Reply:
Might have to do with the fact that they don't really use them for the home feed anymore instead just playing the video when you hover over it?

Reply:
I've moved this into a SimpleVideo object which should unify the API a bit, see commit 87d7bf83cc340a0505c6f5d4c70c1830753f1e68 

It appears as though some other pages return the moving thumbnails more often